### PR TITLE
Removed unneeded persist, remove & merge functions from ServiceReques…

### PR DESCRIPTION
**Pull request by Wrekky**
…tTokenDao. Added a check to manage entities in case removal function is called and entity is not currently in the entity manager.

### CODE REVIEW

File: src/main/java/org/oscarehr/common/dao/ServiceRequestTokenDaoImpl.java [hunk 1]
```diff
@@ -62,7 +47,7 @@ public List<ServiceRequestToken> findAll() {
     @Override
     public ServiceRequestToken findByTokenId(String token) {
         Query query = this.entityManager.createQuery("SELECT x FROM ServiceRequestToken x WHERE x.tokenId = :token", ServiceRequestToken.class);
-        query.setParameter("token", token);
+        query.setParameter("token", token);//
```
Comment (yingbull) [inline src/main/java/org/oscarehr/common/dao/ServiceRequestTokenDaoImpl.java:50]: Is the comment line needed?

File: src/main/java/org/oscarehr/common/dao/ServiceRequestTokenDao.java [hunk 2]
```diff
@@ -86,7 +86,4 @@
 public interface ServiceRequestTokenDao extends AbstractDao<ServiceRequestToken>{
     List<ServiceRequestToken> findAll();
     ServiceRequestToken findByTokenId(String token);
-    void persist(ServiceRequestToken token);
-    void remove(ServiceRequestToken token);
```
Comment (yingbull) [inline src/main/java/org/oscarehr/common/dao/ServiceRequestTokenDao.java:90]: What's the rationale this change? How is it related to the comment on the commit? Feels like it needs to be another commit or at least another comment on the commit message?

**Review by yingbull — CHANGES_REQUESTED** [review]

File: src/main/java/org/oscarehr/common/dao/AbstractDaoImpl.java [hunk 3]
```diff
@@ -114,7 +114,15 @@ public void batchPersist(List<T> oList, int batchSize) {
 	 */
 	@Override
 	public void remove(AbstractModel<?> o) {
-		entityManager.remove(o);
+		//Check if entity is managed
+		boolean isManaged = entityManager.contains(o);
+		AbstractModel<?> managedEntity;
+		if (isManaged) {
+			managedEntity = o;
```
Comment (yingbull) [inline src/main/java/org/oscarehr/common/dao/AbstractDaoImpl.java:121]: I think we need a comment here to make this clearer.  What is it doing when it is managed and why?

File: src/main/java/org/oscarehr/common/dao/ServiceRequestTokenDaoImpl.java [hunk 4]
```diff
@@ -37,21 +37,6 @@ public class ServiceRequestTokenDaoImpl extends AbstractDaoImpl<ServiceRequestTo
...
-}
- 
-@Override
-public void remove(ServiceRequestToken token) {
-    this.entityManager.remove(token);
-}
- 
-@Override
-public ServiceRequestToken merge(ServiceRequestToken token) {
-    return this.entityManager.merge(token);
```
Comment (yingbull) [inline src/main/java/org/oscarehr/common/dao/ServiceRequestTokenDaoImpl.java:53]: Is the rationale for this in Jira? Maybe a hint as to reason for the removals on commit message? I think I follow but could be clearer for others / future.

**Review by yingbull — CHANGES_REQUESTED** [review]

File: src/main/java/org/oscarehr/common/dao/AbstractDaoImpl.java (hunk 3 above)
Comment (Wrekky) [reply src/main/java/org/oscarehr/common/dao/AbstractDaoImpl.java:121]: I'll add a comment after I get the email issue fixed.

File: src/main/java/org/oscarehr/common/dao/ServiceRequestTokenDaoImpl.java (hunk 4 above)
Comment (Wrekky) [reply src/main/java/org/oscarehr/common/dao/ServiceRequestTokenDaoImpl.java:53]: yeahhh you're right, ill add some info about this on the jira ticket and add a comment on removing it in the commit message. This is making it match previous behavior. These functions were previously being called in AbstractDao which this class already extends.

**Pull request closed without merging**